compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/SyntaxTree.java

```diff
@@ -65,29 +65,64 @@ public static SyntaxTree from(SyntaxTree oldTree, TextDocumentChange textDocumen
                 newTextDocument, oldTree.filePath(), false);
     }
 
+    @Deprecated(forRemoval = true)
     public static SyntaxTree asTopLevel(TextDocument textDocument) {
-        return from(ParserRuleContext.TOP_LEVEL_NODE, textDocument);
+        return fromInternal(ParserRuleContext.TOP_LEVEL_NODE, textDocument);
     }
 
+    @Deprecated(forRemoval = true)
     public static SyntaxTree asStatement(TextDocument textDocument) {
-        return from(ParserRuleContext.STATEMENT, textDocument);
+        return fromInternal(ParserRuleContext.STATEMENT, textDocument);
     }
 
+    @Deprecated(forRemoval = true)
     public static SyntaxTree asStatements(TextDocument textDocument) {
-        return from(ParserRuleContext.STATEMENTS, textDocument);
+        return fromInternal(ParserRuleContext.STATEMENTS, textDocument);
     }
 
+    @Deprecated(forRemoval = true)
     public static SyntaxTree asExpression(TextDocument textDocument) {
-        return from(ParserRuleContext.EXPRESSION, textDocument);
+        return fromInternal(ParserRuleContext.EXPRESSION, textDocument);
```

Comment:
This should use `from` right? not `fromInternal`

Reply:
We are going to keep the old API/ behavior intact while introducing the new API which has the balPartNode. So that anything wouldn't break. With the old APIs not just asExpression() is subject to pruning. but also asTopLevel(), asStatement(), asStatements().

Reply:
mmm yeah, I think for #32306 to work they have to use `from` then not `asExpression` right?